File: extract_pdfs.js
const pdf = require('pdf-parse');
const fs = require('fs');
const path = require('path');

const pdfs = [
  'Copy of Copy of MVN Plans.docx.pdf',
  'Copy of Whatsapp Plans.docx.pdf',
  'Exotel Voicebot Commercials.pdf',
  'SIP lines.docx.pdf',
  'SMS Plans.docx.pdf',
  'STD plans.docx.pdf',
  'Toll Free plans.docx.pdf',
  'Truecaller plan.docx.pdf',
  'User based plan.docx.pdf',
  'Veeno user based plans.docx.pdf',
  'Voice streaming.docx.pdf',
];

(async () => {
  for (const file of pdfs) {
    try {
      const buf = fs.readFileSync(path.join(__dirname, file));
      const data = await pdf(buf);
      console.log('\n\n========================================');
      console.log('FILE:', file);
      console.log('========================================');
      console.log(data.text);
    } catch(e) {
      console.log('ERROR reading', file, e.message);
    }
  }
})();

File: extract_pdfs.py
import os
import json
import PyPDF2

pdfs = [
  'Copy of Copy of MVN Plans.docx.pdf',
  'Copy of Whatsapp Plans.docx.pdf',
  'Exotel Voicebot Commercials.pdf',
  'SIP lines.docx.pdf',
  'SMS Plans.docx.pdf',
  'STD plans.docx.pdf',
  'Toll Free plans.docx.pdf',
  'Truecaller plan.docx.pdf',
  'User based plan.docx.pdf',
  'Veeno user based plans.docx.pdf',
  'Voice streaming.docx.pdf',
]

results = {}

for filename in pdfs:
    try:
        with open(filename, 'rb') as file:
            reader = PyPDF2.PdfReader(file)
            text = ''
            for page in reader.pages:
                text += page.extract_text() + '\n'
            # fix spacing issues from extraction
            text = ' '.join([line.strip() for line in text.split('\n') if line.strip()])
            results[filename] = text
    except Exception as e:
        results[filename] = 'ERROR: ' + str(e)

with open('pdf_output.json', 'w', encoding='utf-8') as f:
    json.dump(results, f, indent=2, ensure_ascii=False)
print("Done")
